internal/provider/resource_gitlab_group_project_file_template.go

```diff
@@ -0,0 +1,116 @@
+		Description: `The ` + "`gitlab_group_project_file_template`" + ` resource allows setting a project from which
+custom file templates will be loaded. The project selected must be a direct child of the group identified.
+For more information about which file types are available as templates, view 
+[GitLab's documentation](https://docs.gitlab.com/ee/user/admin_area/settings/instance_template_repository.html#supported-file-types-and-locations)
+
+-> This resource requires a GitLab Enterprise instance with a Premium license.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/groups.html#update-group)`,
+
+		// Since this resource updates an in-place resource, the update method is the same as the create method
+		CreateContext: resourceGitLabGroupProjectFileTemplateCreateOrUpdate,
+		UpdateContext: resourceGitLabGroupProjectFileTemplateCreateOrUpdate,
+		ReadContext:   resourceGitLabGroupProjectFileTemplateRead,
+		DeleteContext: resourceGitLabGroupProjectFileTemplateDelete,
+		// Since this resource updates an in-place resource, importing doesn't make much sense. Simply add the resource
+		// to the config and terraform will overwrite what's already in place and manage it from there.
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Description: `The ID of the group that will use the file template project. This group must be the direct
+                parent of the project defined by project_id`,
+				Type: schema.TypeInt,
+
+				// Even though there is no traditional resource to create, leave "ForceNew" as "true" so that if someone
+				// changes a configuration to a different group, the old group gets "deleted" (updated to have a value
+				// of 0).
+				ForceNew: true,
+				Required: true,
+			},
+			"file_template_project_id": {
+				Description: `The ID of the project that will be used for file templates. This project must be the direct
+				child of the project defined by the group_id`,
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+	}
+})
+
+func resourceGitLabGroupProjectFileTemplateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupId := d.Get("group_id").(int)
+	group, _, err := client.Groups.GetGroup(groupId, nil, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] gitlab group %d not found so removing from state", groupId)
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	if group.MarkedForDeletionOn != nil {
+		log.Printf("[DEBUG] gitlab group %s is marked for deletion", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(fmt.Sprintf("%d", group.ID))
+	d.Set("file_template_project_id", group.FileTemplateProjectID)
+
+	return nil
+}
+
+func resourceGitLabGroupProjectFileTemplateCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupId := d.Get("group_id").(int)
```

Comment:
```suggestion
	groupID := d.Get("group_id").(int)
```

Reply:
refactored throughout the file. Sorry, habit due to naming standards we use at my job 😄